domain/secret/service_test.go

```diff
@@ -56,22 +52,19 @@ func (s *serviceSuite) SetUpTest(c *gc.C) {
 		`, s.modelUUID, coretesting.ControllerTag.Id(), jujuversion.Current.String())
 		return err
 	})
-	s.backendConfigGetter = func(context.Context) (*provider.ModelBackendConfigInfo, error) {
-		return backendConfigs, nil
-	}
 	c.Assert(err, jc.ErrorIsNil)
 }
 
 func (s *serviceSuite) setup(c *gc.C) *gomock.Controller {
 	ctrl := gomock.NewController(c)
-	s.secretBackendReferenceMutator = secret.NewMockSecretBackendReferenceMutator(ctrl)
+	s.secretBackendState = secret.NewMockSecretBackendState(ctrl)
 
 	s.svc = service.NewSecretService(
 		state.NewState(func() (database.TxnRunner, error) { return s.ModelTxnRunner(c, s.modelUUID.String()), nil }, loggertesting.WrapCheckLog(c)),
-		secretbackendstate.NewState(func() (database.TxnRunner, error) { return s.ControllerTxnRunner(), nil }, loggertesting.WrapCheckLog(c)),
+		s.secretBackendState,
 		loggertesting.WrapCheckLog(c),
-		service.SecretServiceParams{BackendAdminConfigGetter: s.backendConfigGetter}).
-		WithBackendRefMutator(s.secretBackendReferenceMutator)
```

Comment:
Why does this suite use both mocks _and_ state?
Why did we add `WithBackendMutator` to switch out what is an argument to the constructor?
Why did we not just add the mock as an argument?
Why is there unused detritus in this file?

Reply:
The mix up between mock and state is clearly a mistake. Good to see it fixed.
The linter should have picked up any unused code - I'm not sure there was any.

Reply:
Note that I have not fixed the real/mocked state bifurcation; only passed the mock directly.